### PR TITLE
Add more transparency into some EC structures.

### PR DESCRIPTION
**Pull request by zhalvorsen**
This adds the ability to create ECDH keys from raw bytes and export signatures as raw bytes.

### CODE REVIEW

File: libraries/crypto/Cargo.toml [hunk 1]
```diff
@@ -25,3 +25,4 @@ zeroize = { version = "1.5.7", features = ["derive"] }
 [features]
 std = ["hex", "ring", "untrusted", "serde", "serde_json", "regex", "rand_core/getrandom"]
 with_ctap1 = []
+acvp = []
```
Comment (kaczmarczyck) [inline libraries/crypto/Cargo.toml]: I don't think we need a feature. Just make anything public without the feature as necessary!

File: libraries/opensk/src/ctap/hid/receive.rs [hunk 2]
```diff
@@ -83,6 +83,7 @@ impl<E: Env> MessageAssembler<E> {
         let (cid, processed_packet) = CtapHid::<E>::process_single_packet(packet);
         if let Some(locked_cid) = locked_cid {
             if locked_cid != cid {
+                #[cfg(not(feature = "acvp"))]
```
Comment (kaczmarczyck) [inline libraries/opensk/src/ctap/hid/receive.rs]: Why is this necessary? This should probably not be here.

File: libraries/crypto/Cargo.toml (hunk 1 above)
Comment (zhalvorsen) [reply libraries/crypto/Cargo.toml]: In both crates?

File: libraries/opensk/src/ctap/hid/receive.rs (hunk 2 above)
Comment (zhalvorsen) [reply libraries/opensk/src/ctap/hid/receive.rs]: Removed

File: libraries/crypto/Cargo.toml (hunk 1 above)
Comment (kaczmarczyck) [reply libraries/crypto/Cargo.toml]: Yes, go ahead! It is just accessors that, if useful for us, might be useful for others.

Comment (zhalvorsen) [reply libraries/crypto/Cargo.toml]: Done

**Review by kaczmarczyck — DISMISSED** [review]

File: libraries/opensk/src/api/crypto/rust_crypto.rs [hunk 3]
```diff
@@ -85,6 +85,11 @@ impl ecdh::SecretKey for SoftwareEcdhSecretKey {
         let shared_secret = self.ephemeral_secret.diffie_hellman(&public_key.public_key);
         SoftwareEcdhSharedSecret { shared_secret }
     }
+
+    fn from_slice(_bytes: &[u8; EC_FIELD_SIZE]) -> Option<Self> {
+        // Currently (8/8/2023) cannot create EphemeralSecret from a slice. Update this, if needed.
+        unimplemented!();
```
Comment (kaczmarczyck) [inline libraries/opensk/src/api/crypto/rust_crypto.rs]: I didn't know that RustCrypto doesn't support this operation. In that case, we probably also shouldn't on this repository.

**Review by kaczmarczyck — COMMENTED** [review]
I have to retract on one of the new functions, sorry for the extra work.

File: libraries/opensk/src/api/crypto/rust_crypto.rs (hunk 3 above)
Comment (zhalvorsen) [reply libraries/opensk/src/api/crypto/rust_crypto.rs]: Removed

**Review by kaczmarczyck — APPROVED** [review]
